contracts/mainnet/connectors/morpho-compound/main.sol

```diff
@@ -0,0 +1,440 @@
+	 * @notice Payback a token to Morpho Compound.
+	 * @param _tokenAddress The address of underlying token to payback.(For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param _poolTokenAddress The address of cToken to payback.(For ETH: cWETH address)
+	 * @param _amount The amount of the token (in underlying) to payback. (For max: `uint256(-1)`)
+	 * @param _getId ID to retrieve amt.
+	 * @param _setId ID stores the amount of tokens paid back.
+	 */
+	function payback(
+		address _tokenAddress,
+		address _poolTokenAddress,
+		uint256 _amount,
+		uint256 _getId,
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		bool _isETH = _tokenAddress == ethAddr;
+		uint256 _amt = getUint(_getId, _amount);
+
+		TokenInterface _tokenContract = _isETH 
+			? TokenInterface(wethAddr) 
+			: TokenInterface(_tokenAddress);
+
+		if (_amt == uint256(-1)) {
+			uint256 _amtDSA = _isETH
+				? address(this).balance
+				: _tokenContract.balanceOf(address(this));
+
+			(, , uint256 _amtDebt) = morphoCompoundLens
+				.getCurrentBorrowBalanceInOf(_poolTokenAddress, address(this));
+
+			_amt = _amtDSA <= _amtDebt ? _amtDSA : _amtDebt;
+		}
+
+		if (_isETH) convertEthToWeth(_isETH, _tokenContract, _amt);
+
+		approve(_tokenContract, address(morphoCompound), _amt);
+
+		morphoCompound.repay(_poolTokenAddress, address(this), _amt);
+
+		setUint(_setId, _amt);
+
+		_eventName = "LogPayback(address,address,uint256,uint256,uint256)";
+		_eventParam = abi.encode(
+			_tokenAddress,
+			_poolTokenAddress,
+			_amt,
+			_getId,
+			_setId
+		);
+	}
+
+	/**
+	 * @dev Payback ETH/ERC20_Token.
+	 * @notice Payback a token to Morpho Compound on behalf of a user.
+	 * @param _tokenAddress The address of underlying token to payback.(For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param _poolTokenAddress The address of cToken to payback.(For ETH: cWETH address)
+	 * @param _onBehalf The address of user who's debt to repay.
+	 * @param _amount The amount of the token (in underlying) to payback. (For max: `uint256(-1)`)
+	 * @param _getId ID to retrieve amt.
+	 * @param _setId ID stores the amount of tokens paid back.
+	 */
+	function paybackOnBehalf(
+		address _tokenAddress,
+		address _poolTokenAddress,
+		address _onBehalf,
+		uint256 _amount,
+		uint256 _getId,
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		bool _isETH = _tokenAddress == ethAddr;
+		uint256 _amt = getUint(_getId, _amount);
+		
+		TokenInterface _tokenContract = _isETH 
+			? TokenInterface(wethAddr) 
+			: TokenInterface(_tokenAddress);
+
+		if (_amt == uint256(-1)) {
+			uint256 _amtDSA = _isETH
+				? address(this).balance
+				: _tokenContract.balanceOf(address(this));
+
+			(, , uint256 _amtDebt) = morphoCompoundLens
+				.getCurrentBorrowBalanceInOf(_poolTokenAddress, _onBehalf);
+
+			_amt = _amtDSA <= _amtDebt ? _amtDSA : _amtDebt;
+		}
```

Comment:
`_amt` is minimumed to the current debt of the user repaid on behalf (https://etherscan.io/address/0x082Bf6702E718483c85423bD279088C215a21302#code#F1#L437) so you can just forward `_amt` to `morphoCompound.repay` in case it's too large

Reply:
Same reason as [here](https://github.com/Instadapp/dsa-connectors/pull/270#discussion_r996240240)